### PR TITLE
Add -Wl,-z,undefs to linux links

### PR DESCRIPTION
**Pull request by keith**
If the user's toolchain / global settings imply -z,defs, this is necessary similar to the macOS flags to allow libpython symbols to be undefined

### CODE REVIEW

**Comment by junyer** [comment]
Should this use [`--allow-shlib-undefined`](https://maskray.me/blog/2020-11-15-explain-gnu-linker-options#allow-shlib-undefined-and---no-allow-shlib-undefined) instead of [`-z undefs`](https://maskray.me/blog/2020-11-15-explain-gnu-linker-options#z-defs-and--z-undefs)? Or even, as the former mentions, `--unresolved-symbols` with one of its `ignore-*` values?

P.S. Your input here would be most welcome, @MaskRay. :)

**Comment by rickeylev** [comment]
Not sure I have enough context to help? I did stumble upon a blog post the other day that might be relevant? https://blog.tim-smith.us/2015/09/python-extension-modules-os-x/

tldr:
* If you're building a python extension module, you want to allow undefined symbols and not link libpython
* If you're embedding python, you want to link libpython

HTH

**Comment by MaskRay** [comment]
> If you're building a python extension module, you want to allow undefined symbols and not link libpython

Then `-Wl,-z,undefs` seems appropriate when there is a global `-z defs`.

**Comment by junyer** [comment]
Okey dokey. Thanks, everyone!

**Comment by keith** [comment]
Thanks all!
